src/java/org/apache/cassandra/streaming/StreamSession.java

```diff
@@ -694,8 +701,8 @@ public Future<?> onError(Throwable e)
             state(State.FAILED); // make sure subsequent error handling sees the session in a final state 
             channel.sendControlMessage(new SessionFailedMessage()).awaitUninterruptibly();
         }
-
-        return closeSession(State.FAILED);
+        StringBuilder boundedThrowable = boundStackTrace(e, 2); //bound the stacktrace with a specified limit on the number of lines
+        return closeSession(State.FAILED, "Failed because of an unkown exception\n" + boundedThrowable);
```

Comment:
Shame to use a StringBuilder and then use a concatenation.
What if you provided the StringBuilder to `boundStackTrace`
```suggestion
        StringBuilder reason = StringBuilder("Failed because of an unknown exception\n");
        boundStackTrace(reason, e, 2); //bound the stacktrace with a specified limit on the number of lines
        return closeSession(State.FAILED, reason.toString());
```

(Also `unkown` typo)